core/app/helpers/spree/base_helper.rb

```diff
@@ -57,8 +57,29 @@ def logo(image_path = nil, options = {})
       end
     end
 
+    def object
+      instance_variable_get('@' + controller_name.singularize)
+    end
+
+    def og_meta_data
+      og_meta = {}
+      if object.is_a? Spree::Product
+        image                             = default_image_for_product_or_variant(object)
+        og_meta['og:image']               = main_app.url_for(image.attachment) if image&.attachment
```

Comment:
not sure if `main_app` or `spree` should be used but this seems to work

Reply:
https://github.com/spree/spree/blob/224fadedf68f7670be913b41b5a1fb33407da095/guides/src/content/release_notes/1_0_0.md#referencing-spree-routes